### PR TITLE
Replace npm with yarn

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Via the [migration doc](https://yarnpkg.com/lang/en/docs/migrating-from-npm/).

So far seems to work fine, is more or less a drop-in replacement. The main thing added in the `yarn.lock` file. [yarn-lock](https://yarnpkg.com/lang/en/docs/yarn-lock/)

### CODE REVIEW

**Comment by rhamilto** [comment]
This smells like change for the sake of change?  And we still need Bower and Grunt, no?

**Comment by benjaminapetersen** [comment]
Bower is deprecated, see #1902. Thats gonna take a decent bit of work to swap over (due to wiredeps, etc).  

NPM for Yarn is very minimal, but will solve some of issues we have with dependencies of dependencies changing on us.  

Grunt is still our task runner.

File: README.md [hunk 1]
```diff
@@ -7,11 +7,10 @@ The management console for [OpenShift Origin](https://github.com/openshift/origi
 Contributing
 ------------
 
-#### Getting started
+#### Getting Started
 
 1. Be sure to have a development environment running for OpenShift. See [the contributing doc](https://github.com/openshift/origin/blob/master/CONTRIBUTING.adoc#develop-locally-on-your-host), we recommend the use of `oc cluster up`.
-1. Install [Nodejs](http://nodejs.org/) and [npm](https://www.npmjs.org/)
-1. Install [grunt-cli](http://gruntjs.com/installing-grunt) and [bower](http://bower.io/) by running `npm install -g grunt-cli bower` (may need to be run with sudo)
```
Comment (rhamilto) [inline README.md]: Grunt and Bower are still needed.

**Comment by spadgett** [comment]
/hold

We should wait until 3.8 to make any changes.

**Comment by benjaminapetersen** [comment]
nice, thats a nifty new flag!

File: README.md (hunk 1 above)
Comment (benjaminapetersen) [reply README.md]: Will fix.

**Pull request closed without merging**